src/please.go

```diff
@@ -456,6 +457,19 @@ var opts struct {
 var buildFunctions = map[string]func() int{
 	"build": func() int {
 		success, state := runBuild(opts.Build.Args.Targets, true, false, false)
+		if success {
+			if opts.Build.OutDir != "" {
+				for _, target := range state.ExpandOriginalLabels() {
+					outs := state.Graph.TargetOrDie(target).FullOutputs()
+					for _, out := range outs {
+						err := fs.CopyFile(out, opts.Build.OutDir+"/"+out, 777)
+						if err != nil {
+							log.Fatalf("Failed to output build to provided directory: %s", err)
+						}
+					}
+				}
+			}
+		}
 		return toExitCode(success, state)
```

Comment:
Small one on code style, in Go people generally prefer to return early rather than nesting the main logic:
```
if !success {
   return toExitCode(success, state)
}
// Logic here so it's not nested another level 
return 0
```

You _could_ arguably also do this for `if opts.Build.OutDir != "" {}` to reduce the nesting there too, but that's more personal preference.